streams/src/main/java/org/apache/kafka/streams/kstream/Suppressed.java

```diff
@@ -155,6 +146,25 @@ static StrictBufferConfig unbounded() {
      * @return a suppression configuration
      */
     static <K> Suppressed<K> untilTimeLimit(final Duration timeToWaitForMoreEvents, final BufferConfig bufferConfig) {
-        return new SuppressedInternal<>(timeToWaitForMoreEvents, bufferConfig, null, false);
+        return new SuppressedInternal<>(null, timeToWaitForMoreEvents, bufferConfig, null, false);
     }
+
+    /**
+     * Use the specified name for the suppression node in the topology and for its changelog.
+     * <p>
+     * This can be used to insert a suppression without changing the rest of the topology names
+     * (and therefore not requiring a reset).
```

Comment:
`a reset` -> `an application reset for upgrading` ?

Reply:
I expanded it to "an application reset".

I'm not sure if "for upgrading" adds any clarity.. it seems ambiguous whether we mean upgrading Streams, or altering the topology, or what.